frontend/packages/dev-console/src/components/import/git/GitSection.tsx

```diff
@@ -100,7 +100,7 @@ const GitSection: React.FC<GitSectionProps> = ({
         values.application.selectedKey !== UNASSIGNED_KEY &&
         setFieldValue('application.name', `${gitRepoName}-app`);
 
-      if (buildStrategy === 'Devfile') {
+      if (buildStrategy === 'Devfile' && !values.devfile?.devfileSourceUrl) {
```

Comment:
Wouldn't this create issues where user adds a git URL and then changes it to something else? This logic would not run the second time and still use the previous devfile contents.

Reply:
I don't think this is an issue, the devfile contents from the user provided git repo are not used anyways,
and the devfile source repo would be updated in the custom hook.

Reply:
I am thinking about the normal Devfile flow where user comes in directly from Add page, in that flow we want to rerun the detection every time user updates the git repo URL. But the custom hook only runs for git repo URL coming from query argument in case of sample or catalog flow.

Reply:
In the normal devfile flow, we aren't using another repo for the devfile source, so `values.devfile.devfileSourceUrl` would not be set.

Reply:
Ah, so its not being used anywhere else except in catalog flow. Makes sense.